x/observer/types/genesis.go

```diff
@@ -38,13 +38,13 @@ func (gs GenesisState) Validate() error {
 	}
 
 	// Check for duplicated index in chainNonces
-	chainNoncesIndexMap := make(map[string]bool)
+	chainNoncesIndexMap := make(map[int64]bool)
```

Comment:
nit: in Go you can use a map of `struct{}` to allocate less memory. In particular no memory is allocated for map values at all. Example: `map[int64]struct{}{}`

Reply:
[85eef7c](https://github.com/zeta-chain/node/pull/2515/commits/85eef7c5c3086048fdb1c1dc64cb0676f6ef1d1c)